extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,149 @@
+from conan.cli.args import common_graph_args, validate_common_graph_args
+from conan.cli.command import conan_command
+
+if typing.TYPE_CHECKING:
+    from conans.client.graph.graph import Node
+
+
+def package_type_to_component_type(pt: str) -> 'ComponentType':
+    if pt is "application":
+        return ComponentType.APPLICATION
+    else:
+        return ComponentType.LIBRARY
+
+
+def licenses(ids):
+    """
+    see https://cyclonedx.org/docs/1.4/json/#components_items_licenses
+    """
+    if ids is None:
+        return None
+    if not isinstance(ids, tuple):
+        ids = [ids]
+    return [LicenseChoice(license=LicenseFactory().make_from_string(i)) for i in ids]
+
+
+def name(n: 'Node', aux_vars: dict) -> str:
+    if n.name:
+        return n.name
+    else:
+        assert aux_vars["unknown_name_created"] is False, "multiple nodes have no name"
+        aux_vars["unknown_name_created"] = True
+        return "UNKNOWN"
+
+
+def package_url(node: 'Node', cached_name: str) -> 'PackageURL':
+    """
+    Creates a PURL following https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#conan
+    """
+    return PackageURL(
+        type="conan",
+        name=cached_name,
+        version=node.conanfile.version,
+        qualifiers={
+            "prev": node.prev,
+            "rref": node.ref.revision if node.ref else None,
+            "user": node.conanfile.user,
+            "channel": node.conanfile.channel,
+            "repository_url": node.remote.url if node.remote else None
+        })
+
+
+def create_component(n: 'Node', aux_vars: dict) -> 'Component':
+    name_ = name(n, aux_vars)
+    purl = package_url(n, name_)
+    result = Component(
+        type=package_type_to_component_type(n.conanfile.package_type),
+        name=name_,
+        version=n.conanfile.version,
+        licenses=licenses(n.conanfile.license),
+        bom_ref=purl.to_string(),
+        purl=purl,
+        description=n.conanfile.description
+    )
+    if n.conanfile.homepage:
+        result.external_references.add(ExternalReference(
+            type=ExternalReferenceType.WEBSITE,
+            url=XsUri(n.conanfile.homepage),
+        ))
+    return result
+
+
+def format_cyclonedx_14_json(result):
+    serialized_json = JsonV1Dot4(result).output_as_string()
+    cli_out_write(serialized_json)
+
+
+def format_text(result):
+    out = ConanOutput()
+    out.error("Format 'text' not supported")
+
+
+@conan_command(group="Recipe", formatters={
+    "text": format_text,  # added by default in BaseConanCommand.__init__
+    "cyclonedx_1.4_json": format_cyclonedx_14_json})
+def create_sbom(conan_api: ConanAPI, parser, *args):
+    """
+    creates an SBOM in CycloneDX 1.4 JSON format
+    """
+    for package_name, found in found_libs.items():
+        if not found:
+            sys.stderr.write(
+                f"The sbom extension needs an additional package, please run 'pip install {package_name}'\n")
+            sys.exit(1)
```

Comment:
It is a simpler to make the imports with try-except local. It is not compulsory that imports are in the global scope, they can be local to functions.

Reply:
imports are local now